server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java

```diff
@@ -835,4 +837,34 @@ public void testExpireAt() {
       assertThat(redis.expireat(k(1), timeService.wallClockTime() + 1000, ExpireArgs.Builder.xx())).isFalse();
       assertThat(redis.expireat(k(1), timeService.wallClockTime() + 1000, ExpireArgs.Builder.nx())).isTrue();
    }
+
+   @Test
+   public void testPFADD() {
+      RedisCommands<String, String> redis = redisConnection.sync();
+
+      // Just creates the new structure. But Lettuce does not accept an empty argument.
+      // assertThat(redis.pfadd("my-hll")).isEqualTo(1L);
+
+      assertThat(redis.pfadd("my-hll", "el1", "el2", "el3"))
+            .isEqualTo(1L);
+
+      assertThat(redis.pfadd("my-hll", "el1")).isEqualTo(0L);
+
+      // Until 193 it is using the explicit representation. It could only return false in case of a hash conflict.
+      for (int i = 4; i < 193; i++) {
+         assertThat(redis.pfadd("my-hll", "el" + i)).isEqualTo(1L);
+      }
+
+      // From this point on, it is using the probabilistic estimation.
+      SoftAssertions sa = new SoftAssertions();
+      for (int i = 193; i < 1024; i++) {
+         sa.assertThat(redis.pfadd("my-hll", "el" + i)).isEqualTo(1L);
+      }
```

Comment:
this tests pfadd of distinct elements, does it worth to add the same test for pfadd of replicated entries?

Reply:
Yeah, let me update.

Reply:
Updated. The test tries to insert duplicated elements with explicit and compact sets and expects a return of 0. I also added a test in `HyperLogLogTests` to check for duplicated elements.